### PR TITLE
Option to use Gen 3 Physical/Special split.

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Option in `battle_config.h` to use Gen 3's physical/special split.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

File: include/constants/battle_config.h [hunk 1]
```diff
@@ -72,6 +72,7 @@
 #define B_PSYWAVE_DMG               GEN_6 // Psywave's damage formula. See Cmd_psywavedamageeffect.
 
 // Move settings
+#define B_PHYSICAL_SPECIAL_SPLIT    GEN_3 // In Gen3, the move's type determines if it will do physical or special damage.
```
Comment (ExpoSeed) [inline include/constants/battle_config.h:75]: The default should be greater than `GEN_3` for consistency with the rest of the default options and to not accidentally revert the split in existing hacks unexpectedly

File: src/battle_main.c [hunk 2]
```diff
@@ -4288,7 +4288,7 @@ s8 GetMovePriority(u32 battlerId, u16 move)
         priority++;
     }
     else if (GetBattlerAbility(battlerId) == ABILITY_PRANKSTER
-        && gBattleMoves[move].split == SPLIT_STATUS)
+        && GetBattleMoveSplit(move) == SPLIT_STATUS)
```
Comment (ExpoSeed) [inline src/battle_main.c:4291]: Use `IS_MOVE_STATUS` now that you have created it

File: src/battle_util.c [hunk 3]
```diff
@@ -4247,7 +4247,7 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u8 ability, u8 special, u16 moveA
             if (!(gMoveResultFlags & MOVE_RESULT_NO_EFFECT)
              && TARGET_TURN_DAMAGED
              && IsBattlerAlive(battler)
-             && gBattleMoves[gCurrentMove].split == SPLIT_PHYSICAL
+             && GetBattleMoveSplit(gCurrentMove) == SPLIT_PHYSICAL
```
Comment (ExpoSeed) [inline src/battle_util.c:4250]: Use `IS_MOVE_PHYSICAL`

File: src/pokemon_summary_screen.c [hunk 4]
```diff
@@ -3708,7 +3708,7 @@ static void PrintMoveDetails(u16 move)
     {
         if (sMonSummaryScreen->currPageIndex == PSS_PAGE_BATTLE_MOVES)
         {
-            ShowSplitIcon(gBattleMoves[move].split);
+            ShowSplitIcon(GetBattleMoveSplit(move));
```
Comment (ExpoSeed) [inline src/pokemon_summary_screen.c:3711]: Question: should this show an icon if the split is disabled? It is more useful to the player if the icon is there, but it is more faithful to Gen 3 if the icon is disabled along with the split.

**Review by ExpoSeed — CHANGES_REQUESTED** [review]
Nice feature

File: src/pokemon_summary_screen.c (hunk 4 above)
Comment (AsparagusEduardo) [reply src/pokemon_summary_screen.c]: Added option to hide the split icon independent from the split mechanics.

**Review by ExpoSeed — DISMISSED** [review]
thanks for cleaning the battle config comments too, that was bugging me

File: src/battle_ai_script_commands.c [hunk 5]
```diff
@@ -2801,19 +2801,19 @@ static void Cmd_get_curr_dmg_hp_percent(void)
...
 {
-    AI_THINKING_STRUCT->funcResult = gBattleMoves[AI_THINKING_STRUCT->moveConsidered].split;
+    AI_THINKING_STRUCT->funcResult = GetBattleMoveSplit(AI_THINKING_STRUCT->moveConsidered);
     gAIScriptPtr += 1;
 }
 
 static void Cmd_get_considered_move_target(void)
 {
-    AI_THINKING_STRUCT->funcResult = gBattleMoves[AI_THINKING_STRUCT->moveConsidered].target;
+    AI_THINKING_STRUCT->funcResult = GetBattleMoveSplit(AI_THINKING_STRUCT->moveConsidered);
```
Comment (ultima-soul) [inline src/battle_ai_script_commands.c:2816]: I think this change may be an error?

**Review by ultima-soul — CHANGES_REQUESTED** [review]
Functionality-wise seem good.

File: src/battle_ai_script_commands.c (hunk 5 above)
Comment (AsparagusEduardo) [reply src/battle_ai_script_commands.c:2816]: I think it is, yeah xD

File: include/constants/battle_config.h [hunk 6]
```diff
@@ -97,29 +98,32 @@
...
 
-// Other
-#define B_FLAG_INVERSE_BATTLE       0     // If this flag is set, the battle's type effectiveness are inversed. For example, fire is super effective against water. 0 disables the feature.
+// Interface settings
 #define B_FAST_INTRO                TRUE  // If set to TRUE, battle intro texts print at the same time as animation of a Pokémon, as opposing to waiting for the animation to end.
 #define B_SHOW_TARGETS              TRUE  // If set to TRUE, all available targets, for moves hitting 2 or 3 Pokémon, will be shown before selecting a move.
+#define B_SHOW_SPLIT_ICON           TRUE  // If set to TRUE, it will show an icon in the summary showing the move's category split.
+
+// Other
+#define B_FLAG_INVERSE_BATTLE       0     // If this flag is set, the battle's type effectiveness are inversed. For example, fire is super effective against water. 0 disables the feature.
```
Comment (ultima-soul) [inline include/constants/battle_config.h:107]: Clarify this comment to mention that this is the flag ID you're assigning to use in scripting, etc.

**Review by ExpoSeed — APPROVED** [review]

**Review by ultima-soul — APPROVED** [review]

**Pull request merged**